csharp/example/Program.cs

```diff
@@ -0,0 +1,134 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using Microsoft.Data.Analysis;
+using Grpc.Net.Client;
+using Grpc.Core;
+using Apache.Arrow.Flight.Client;
+using Apache.Arrow.Flight;
+using Apache.Arrow;
+
+namespace FlightClientExample
+{
+    public class Program
+    {
+        public static async Task Main(string[] args)
+        {
+            string host = "localhost";
+            string port = "32010";
+            string user = "dremio";
+            string pass = "dremio123";
+            // For default query, add the Samples source and Format the zips.json in Dremio UI
+            string query = "SELECT city, loc[0] AS longitude, loc[1] AS latitude, pop, state, _id FROM Samples.\"samples.dremio.com\".\"zips.json\" LIMIT 100";
+            string protocol = "http";
+
+            // Parse command-line arguments to override the defaults when set
+            for (int i = 0; i < args.Length-1; i++)
+            {
+                if (i % 2 == 0 && args[i].StartsWith("-")) 
+                {
+                    var key = args[i];
+                    var value = args[i+1];
+                    if (key.EndsWith("host")) host = value;
+                    if (key.EndsWith("port")) port = value;
+                    if (key.EndsWith("user")) user = value;
+                    if (key.EndsWith("pass")) pass = value;
+                    if (key.EndsWith("query")) query = value;
+                    if (key.EndsWith("protocol")) protocol = value;
+                }
+            }
+
+            // Basic auth using username and password
+            string encoded = System.Convert.ToBase64String(System.Text.Encoding.GetEncoding("ISO-8859-1").GetBytes(user + ":" + pass));
+            Console.WriteLine($"The encoded credentials: {encoded}");
+
+            // Create client
+            var address = $"{protocol}://{host}:{port}";
+            Console.WriteLine($"Connecting to: {address}");
+
+            var handler = new HttpClientHandler();
+            // For localhost https (TLS) endpoint testing, uncomment the following to avoid a cert error
+            // handler.ServerCertificateCustomValidationCallback = HttpClientHandler.DangerousAcceptAnyServerCertificateValidator;
+
+            var httpClient = new HttpClient(handler);
+            httpClient.DefaultRequestHeaders.Add("Authorization", "Basic " + encoded);
+
+            // An example header for token authentication instead of Basic auth
+            // httpClient.DefaultRequestHeaders.Add("Authorization", "Bearer " + "X4NxSDN5...H11kUqYU/vWmzA==");
+           
+            var channel = GrpcChannel.ForAddress(address, new GrpcChannelOptions
+            {
+                HttpClient = httpClient
+            });
+
+            FlightClient client = new FlightClient(channel);
+
+            // Pass the query text as the Command Descriptor

```

Comment:
Creating a separate method for this would modularise and make the code more legible.